### DISCUSSION TITLE
IPV6异常

### DISCUSSION THREAD

**Discussion by Danny-Ken** [kkkgo/PaoPaoDNS#133] [问答和帮助]
IPV6配置为yes时，我记得以前的版本境外域名只返回ipv4的结果，但是刚我把cloudflare.com放到nocn列表后，我尝试通过容器测试，居然返回了IPV6的结果，测试几个境外域名都是这样(以前只返回IPV4的域名)，目前是更新的4-30版本

**Comment by kkkgo** [comment]
没有问题，也许你的测试命令不对，或者返回的结果并不是PaoPaoDNS提供的。
你可以开启ADDINFO用dig命令观察结果来源。

**Reply by Danny-Ken** [reply]
一直都开着ADDINFO，以下是返回的信息

/ # dig AAAA www.cloudflare.com

; <<>> DiG 9.18.25 <<>> AAAA www.cloudflare.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 561
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 2

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;www.cloudflare.com.            IN      AAAA

;; ANSWER SECTION:
www.cloudflare.com.     0       IN      AAAA    2606:4700::6810:7b60
www.cloudflare.com.     0       IN      AAAA    2606:4700::6810:7c60

;; ADDITIONAL SECTION:
20240503130846.035906624.addinfo.paopaodns. 1 IN TXT "Since 1ms From:nocn forward_unbound"

;; Query time: 0 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Fri May 03 05:08:46 UTC 2024
;; MSG SIZE  rcvd: 229

**Reply by Danny-Ken** [reply]
在另一台设备全新部署后也是一样的情况，没有添加域名在force_dnscrypt_list的时候，返回的是FakeIP，添加后返回的就有IPV6地址

**Reply by Danny-Ken** [reply]
对了，IPV6的环境变量是yes_only6

**Reply by kkkgo** [reply]
带only6的需要先对A记录进行探测。缺失了探测步骤。你拉一下新镜像试试。

**Reply by Danny-Ken** [reply]
拉取新的镜像OK，感谢大佬的付出，谢谢
